src/docgen/view/documentation.ts

```diff
@@ -1,121 +1,227 @@
+import * as child from 'child_process';
+import * as os from 'os';
+import * as path from 'path';
+import * as util from 'util';
+import * as fs from 'fs-extra';
+import * as glob from 'glob-promise';
 import * as reflect from 'jsii-reflect';
+import { TargetLanguage } from 'jsii-rosetta';
+import { transliterateAssembly } from 'jsii-rosetta/lib/commands/transliterate';
 import { Markdown } from '../render/markdown';
 import { PythonTranspile } from '../transpile/python';
 import { Transpile } from '../transpile/transpile';
 import { TypeScriptTranspile } from '../transpile/typescript';
 import { ApiReference } from './api-reference';
 import { Readme } from './readme';
 
+const exec = util.promisify(child.exec);
+
 /**
- * Options for rendering documentation pages.
+ * Options for rendering a `Documentation` object.
  */
-export interface DocumentationOptions {
+export interface RenderOptions {
+
   /**
-   * The assembly to generate from.
+   * Include a generated api reference in the documentation.
+   *
+   * @default true
    */
-  readonly assembly: reflect.Assembly;
+  readonly apiReference?: boolean;
 
   /**
-   * Which language to generate docs for.
-   */
-  readonly language: string;
+    * Include the user defined README.md in the documentation.
+    *
+    * @default true
+    */
+  readonly readme?: boolean;
 
   /**
-   * Include a generated api reference in the documentation.
+    * Generate documentation only for a specific submodule.
+    *
+    * @default - Documentation is generated for the root module only.
+    */
+  readonly submodule?: string;
+
+}
+
+/**
+ * Options for creating a `Documentation` object.
+ */
+export interface DocumentationOptions {
+
+  /**
+   * Which language to generate docs for.
    *
-   * @default true
+   * @default 'ts'
    */
-  readonly apiReference?: boolean;
+  readonly language?: string;
 
   /**
-   * Include the user defined README.md in the documentation.
+   * Whether to ignore missing fixture files that will prevent transliterating
+   * some code snippet examples.
    *
    * @default true
    */
-  readonly readme?: boolean;
+  readonly loose?: boolean;
+}
+
+/**
+ * Options for creating a `Documentation` object using the `fromLocalPackage` function.
+ */
+export interface FromLocalPackageDocumentationOptions extends DocumentationOptions {
 
   /**
-   * Generate documentation only for a specific submodule.
+   * A local directory containing jsii assembly files that will
+   * comprise the type-system.
    *
-   * @default - Documentation is generated for all submodules.
+   * @default - the root package directory will be used.
```

Comment:
What does this mean? I'm not sure I get it...

Reply:
Yeah, its kind of tricky...

This refers to the required argument in the `forLocalPackage` function. It means that all assemblies are expected to be available in the root directory (normally `<package-dir>/node_modules).

You feel this needs further explanation here? it's one of those properties that I hope don't get used much, and if they are, than its because the user is looking for this support (like we do in `forRemotePackage`).